### ISSUE TITLE
delve prints ←[34m in Windows command window

### ISSUE THREAD

**Issue by alexbrainman** [derekparker/delve#436]
CC @lukehoban

To reproduce do this:
```
C:\dev\src\t>systeminfo | findstr /B /C:"OS Name" /C:"OS Version"
OS Name:                   Microsoft Windows 7 Professional
OS Version:                6.1.7601 Service Pack 1 Build 7601

C:\dev\src\t>go version
go version devel +a4b8339 Mon Feb 22 21:15:50 2016 +0000 windows/amd64

C:\dev\src\t>dlv version
Delve Debugger
Version: 0.11.0-alpha
Build:

C:\dev\src\t>type main.go
package main

import (
        "fmt"
)

func main() {
        for i := 0; i < 10; i++ {
                fmt.Printf("i=%d\n", i)
        }
}

C:\dev\src\t>dlv debug
Type 'help' for list of commands.
(dlv) b main.main
Breakpoint 1 set at 0x40105f for main.main() c:/dev/src/t/main.go:7
(dlv) c
> main.main() c:/dev/src/t/main.go:7 (hits goroutine(1):1 total:1) (PC: 0x40105f)
←[34m     2:    ←[0m
←[34m     3:    ←[0mimport (
←[34m     4:    ←[0m    "fmt"
←[34m     5:    ←[0m)
←[34m     6:    ←[0m
←[34m=>   7:    ←[0mfunc main() {
←[34m     8:    ←[0m    for i := 0; i < 10; i++ {
←[34m     9:    ←[0m            fmt.Printf("i=%d\n", i)
←[34m    10:    ←[0m    }
←[34m    11:    ←[0m}
(dlv) q

C:\dev\src\t>
```

I see that delve determines wether to print escape codes or not by checking `strings.ToLower(os.Getenv("TERM")) == "dumb"`. Perhaps it is OK on non Windows OS. But Windows command line output doesn't do escape codes. Windows also normally does not set TERM environment variable. I propose to change windows code to assume terminal is "dumb" also if TERM environment variable is not found or empty. What do you think? Alternatives? Should we change this code for every OS?

Alex

**Comment by jussij** [comment]
>  But Windows command line output doesn't do escape codes.

I'm running Windows 10 and it is handling those escape codes correctly.

**Comment by lukehoban** [comment]
Yeah - I hadn't seen this myself since I'd been on Win10 builds, but recently did see this on a Win2k8 server VM during some testing.

I'm not sure what the best approach is here.  I wouldn't want to lose the escape codes for Win10 or ConEmu users, but I don't know of any way to test whether the console host supports these to reliably suppress them in pre-Win10 conhost environments.

**Comment by alexbrainman** [comment]
As I understand it every version of Windows before Windows 10 is affected. I think we should disable  printing escape codes for all users running on Windows versions before Windows 10 - output colour is not as important as being able to understand what your program says. Disable printing escape codes by default. Perhaps we should allow some way for user to control if delve outputs escape codes or not, but I am not sure TERM environment variable is a good way. For example, my version of git requires that I set TERM=msys. So I am kind of stuck with that.

There would be nice to be able to determine if Windows console supports outputs escape codes somehow. But I also don't know how to do that.

Alex

**Comment by jhowardmsft** [comment]
@alexbrainman Alex - it is possible to infer whether or not the console being used supports ANSI sequences. We do this in docker. For example, https://github.com/docker/docker/blob/master/pkg/term/term_windows.go#L40-L55 looks at various environment variables. In addition, see the code in https://github.com/docker/docker/blob/master/pkg/term/term_windows.go#L103-L173 for the *NativeConsole functions which will determine programmatically whether or not the conhost Windows console is capable of supporting ANSI sequences natively. That probing/setting should work on all versions of Windows (well at least 7+). Note that docker currently does NOT enable the native console yet, which is why there's that OS version checking in there https://github.com/docker/docker/blob/master/pkg/term/term_windows.go#L72 which should not be necessary in this case.

Hope that helps
John.

**Comment by jhowardmsft** [comment]
Should have added, the probing is necessary as on Windows 10, it is still possible to turn off the new ANSI-capable console support through this checkbox on a cmd windows properties:

![consolemode](https://cloud.githubusercontent.com/assets/10522484/13380778/4c427d66-ddff-11e5-8346-cf457ab0d8e8.JPG)

**Comment by alexbrainman** [comment]
I will have a look at your code @jhowardmsft . Thank you.

Alex

**Comment by alexbrainman** [comment]
I tried following @jhowardmsft suggestion (not completly), and I got this https://github.com/alexbrainman/delve/commit/3fdfa4e164c50ed61e550196e2a175a70fa426a9 . But I don't have Windows 10 computer to test it. @jussij and @lukehoban would you mind trying my change?

Thank you.

Alex

**Comment by lukehoban** [comment]
@alexbrainman On my Windows 10 laptop, I'm still seeing the console colors as expected.  When I turn on the "Use legacy console" setting that @jhowardmsft shows in the cmd.exe properties UI, I no longer see colorization.  So this looks good to me.

**Comment by jhowardmsft** [comment]
Yay :)

**Comment by alexbrainman** [comment]
Thanks for checking. I have create a PR.

Alex

**Comment by aarzilli** [comment]
Fixed by ff0ec8ce00fbac7f84859cf420b31bc3a513e43c
